Taken from Oracle solaris-userland b83a793,
components/x11/xserver/xorg/patches/16-dixmods-deps.patch (xorg-server 21.1.24).

Solaris specific build patches

not suitable for upstream

--- xorg-server-21.1.20/hw/xfree86/Makefile.am
+++ xorg-server-21.1.20/hw/xfree86/Makefile.am
@@ -139,7 +139,7 @@ CLEANFILES = sdksyms.c sdksyms.dep Xorg.
 EXTRA_DIST += sdksyms.sh
 
 sdksyms.dep sdksyms.c: sdksyms.sh
-	$(AM_V_GEN)CPP='$(CPP)' AWK='$(AWK)' $(SHELL) $(srcdir)/sdksyms.sh $(top_srcdir) $(CFLAGS) $(AM_CFLAGS) $(AM_CPPFLAGS)
+	$(AM_V_GEN)CPP='$(CPP)' AWK='$(AWK)' $(SHELL) $(srcdir)/sdksyms.sh $(top_srcdir) $(CFLAGS) $(AM_CFLAGS) $(AM_CPPFLAGS) -I/usr/include -I/usr/include/xorg -I/usr/include/pixman-1 -I/usr/include/X11/dri
 
 SDKSYMS_DEP = sdksyms.dep
 -include $(SDKSYMS_DEP)
--- xorg-server-21.1.20/configure.ac
+++ xorg-server-21.1.20/configure.ac
@@ -198,6 +198,8 @@ fi
 dnl Find the math library, then check for cbrt function in it.
 AC_CHECK_LIB(m, sqrt)
 AC_CHECK_FUNCS([cbrt])
+# But we don't want to link everything with libm
+LIBS="${LIBS//-lm}"
 
 dnl AGPGART headers
 AC_ARG_ENABLE(agp, AS_HELP_STRING([--enable-agp],
@@ -1664,6 +1666,7 @@ XSERVER_CFLAGS="${XSERVER_CFLAGS} ${XSER
 XSERVER_LIBS="$DIX_LIB $MI_LIB $OS_LIB"
 XSERVER_SYS_LIBS="${XSERVERLIBS_LIBS} ${SYS_LIBS} ${LIBS}"
 XSERVER_SYS_LIBS="${XSERVER_SYS_LIBS} -lproject"
+XSERVER_SYS_LIBS="${XSERVER_SYS_LIBS} -lm"
 AC_SUBST([XSERVER_LIBS])
 AC_SUBST([XSERVER_SYS_LIBS])
 
